src/librustdoc/scrape_examples.rs

```diff
@@ -139,26 +142,41 @@ where
         }
 
         // Get type of function if expression is a function call
-        let (ty, span) = match ex.kind {
+        let (ty, call_span, ident_span) = match ex.kind {
             hir::ExprKind::Call(f, _) => {
                 let types = tcx.typeck(ex.hir_id.owner);
 
                 if let Some(ty) = types.node_type_opt(f.hir_id) {
-                    (ty, ex.span)
+                    (ty, ex.span, f.span)
                 } else {
                     trace!("node_type_opt({}) = None", f.hir_id);
                     return;
                 }
             }
-            hir::ExprKind::MethodCall(_, _, span) => {
+            hir::ExprKind::MethodCall(_, args, call_span) => {
                 let types = tcx.typeck(ex.hir_id.owner);
                 let def_id = if let Some(def_id) = types.type_dependent_def_id(ex.hir_id) {
                     def_id
                 } else {
                     trace!("type_dependent_def_id({}) = None", ex.hir_id);
                     return;
                 };
-                (tcx.type_of(def_id), span)
+
+                // The MethodCall node doesn't directly contain a span for the
+                // method identifier, so we have to compute it by trimming the full
+                // span based on the arguments.
+                let ident_span = match args.get(1) {
+                    // If there is an argument, e.g. "f(x)", then
+                    // get the span "f(" and delete the lparen.
+                    Some(arg) => {
+                        let with_paren = call_span.until(arg.span);
+                        with_paren.with_hi(with_paren.hi() - BytePos(1))
+                    }
+                    // Otherwise, just delete both parens directly.
+                    None => call_span.with_hi(call_span.hi() - BytePos(2)),
+                };
```

Comment:
I'm pretty sure you can just use the first field of `MethodCall`. It's the path segment for the method name. Also, this Span-manipulating code will break with macros.

Reply:
Ah, I was confused b/c `PathSegment` doesn't have a span. But I see that `PathSegment::ident` has a span. Should I just use that, and ignore stuff like type parameters?

Reply:
I think ignoring type parameters sounds good. So, yeah, just use the ident's span.